kong/db/schema/entities/routes.lua

```diff
@@ -43,6 +43,11 @@ return {
                          len_min  = 1,
                          required = true,
                          elements = typedefs.protocol,
+                         mutually_exclusive_subsets = {
```

Comment:
I don't think the `mutually_exclusive_subsets` validator is worth its complexity, if it's used only in this entitity. I would have used a custom validator and done the checks manually with 3 ifs.

Reply:
This pattern has appeared in the past and will probably appear again. I discussed it with @gszr and we agreed adding the validator was valuable — certainly not worth redesigning this at this point.

Reply:
Ok, marking this as resolved then.